pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -493,3 +511,10 @@ func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet)
 func (m *manager) GetCPUs(podUID, containerName string) cpuset.CPUSet {
 	return m.state.GetCPUSetOrDefault(podUID, containerName)
 }
+
+func (m *manager) updateAdmittedPods(pod *v1.Pod) {
+	m.Lock()
+	defer m.Unlock()
+
+	m.pendingAdmissionPod = pod
+}
```

Comment:
Do we actually even need this function anymore (in any of the components)? Unless I'm missing something, no one should be calling this asynchrounously so no lock should be necesssary. If it is still necessary, we should rename it to  something more appropriate (e.g. `setPodPendingAdmission()`).

Reply:
I think we need this one at least for the CPU manager and device manager:
1. CPU manager - we have the reconcile loop that can happen in the middle of the other methods
2. Device manager - it possible to call pod resource API that calls to `UpdateAllocatedDevices` during other device managers methods

Reply:
Both the reconcile loop and `UpdateAllocatedDevices ()` should not be updating this variable though (only consuming it).

Reply:
Renamed.

Reply:
But it is not an atomic variable so if you will read it in the middle of the write operation you can get some unexpected results(never encounter it, but I know it possible in theory)

Reply:
It’s a pointer, so it should be OK (on most systems), but I agree it is not guaranteed in general. Are all reads protected by the same lock used for the writes?

Reply:
Yes, we are using the same lock for the whole manager.